walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -19,39 +19,37 @@
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    ArrayList<Event> eventsArrayList = new ArrayList<Event>();
     Collection<TimeRange> meetingTimes = new ArrayList<>();
 
     if(request.getDuration() > TimeRange.WHOLE_DAY.duration()){
         return meetingTimes;
     }
     
+    ArrayList<TimeRange> conflictingEventTimes = new ArrayList<TimeRange>();
     for(Event event: events){
         if(checkAttending(event, request)){
-            eventsArrayList.add(event);
+            conflictingEventTimes.add(event.getWhen());
         }
     }
-
-    if(eventsArrayList.size() == 0){
+    
+    if(conflictingEventTimes.size() == 0){
         meetingTimes.add(TimeRange.WHOLE_DAY);
         return meetingTimes;
     }
 
-    //sort by begin time
-    //eventsArrayList could be more specific 
-    //initialize > populate, not initialize > initilaize > populate
+    Collections.sort(conflictingEventTimes, TimeRange.ORDER_BY_START);
 
     //check first event start time against beginning of day
-    int firstEventStart = eventsArrayList.get(0).getWhen().start();
+    int firstEventStart = conflictingEventTimes.get(0).start();
     if((firstEventStart - TimeRange.START_OF_DAY) >= request.getDuration()){
         meetingTimes.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, firstEventStart, false));
     }
 
-    int latestEndTime = eventsArrayList.get(0).getWhen().end();
+    int latestEndTime = conflictingEventTimes.get(0).end();
     //checks all middle events to see if there is enough time between them
```

Comment:
Ooh, I just realized this is a bug. The event with the latest start time may not have the latest end time.

Consider also adding a test case for this.

Reply:
I'm comparing latest end times, not start times. That's why I track the end time w another variable. I can add a test case for it still, though.